### PR TITLE
Release notes v4.0.0

### PR DESCRIPTION
**Pull request by johnnypham**

### CODE REVIEW

File: release-notes/4.0/4.0.0.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+- System.Runtime.Caching 5.0.0
+- System.Text.Encoding.CodePages 5.0.0
+- System.Text.Encodings.Web 4.7.2
+- System.Resources.ResourceManager 4.3.0
+- System.Security.Cryptography.Cng 5.0.0
+- System.Security.Principal.Windows 5.0.0
+
+#### .NET Standard
+
+- Microsoft.Data.SqlClient.SNI.runtime 4.0.0-preview1.21232.1
```
Comment (JRahnama) [inline release-notes/4.0/4.0.0.md:169]: I think this should be changed to GA version

File: release-notes/4.0/4.0.0.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+- .NET Framework 4.6.1+ (Windows x86, Windows x64)
+- .NET Core 3.1+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+- .NET Standard 2.0+ (Windows x86, Windows x64, Windows ARM64, Windows ARM, Linux, macOS)
+
+### Dependencies
+
+#### .NET Framework
+
+- Microsoft.Data.SqlClient.SNI 4.0.0-preview1.21232.1
```
Comment (JRahnama) [inline release-notes/4.0/4.0.0.md:135]: GA version

File: release-notes/4.0/4.0.0.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+- System.Configuration.ConfigurationManager 5.0.0
+- System.IO 4.3.0
+- System.Runtime.InteropServices.RuntimeInformation 4.3.0
+- System.Security.Cryptography.Algorithms 4.3.1
+- System.Security.Cryptography.Primitives 4.3.0
+- System.Text.Encodings.Web 4.7.2
+
+#### .NET Core
+
+- Microsoft.Data.SqlClient.SNI.runtime 4.0.0-preview1.21232.1
```
Comment (JRahnama) [inline release-notes/4.0/4.0.0.md:150]: same here

**Comment by cheenamalhotra** [comment]
I think you also need to update Changelog.md with changes summarized between preview3 and now. Also include the same in Release notes (check old GA release notes for reference).

**Comment by johnnypham** [comment]
> I think you also need to update Changelog.md with changes summarized between preview3 and now. Also include the same in Release notes (check old GA release notes for reference).

I don't think there were any changes that needed to be included. However, once the new version of SNI is released, I can add that.

File: release-notes/4.0/4.0.0.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Encrypt default value set to true
+The default value of the `Encrypt` connection setting has been changed from `false` to `true`. With the growing use of cloud databases and the need to ensure those connections are secure, it's time for this backwards-compatibility-breaking change.
+
+### Ensure connections fail when encryption is required
+In scenarios where client encryption libraries were disabled or unavailable, it was possible for unencrypted connections to be made when Encrypt was set to true or the server required encryption.
+
+###  App Context Switch for using System default protocols
+TLS 1.3 is not supported by the driver; therefore, it has been removed from the supported protocols list by default. Users can switch back to forcing use of Operating System's client protocols, by enabling the App Context switch below:
```
Comment (David-Engel) [inline release-notes/4.0/4.0.0.md:72]:
```suggestion
TLS 1.3 is not supported by the driver; therefore, it has been removed from the supported protocols list by default. Users can switch back to forcing use of the Operating System's client protocols, by enabling the App Context switch below:
```

File: release-notes/4.0/4.0.0.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+ `Switch.Microsoft.Data.SqlClient.UseSystemDefaultSecureProtocols`
+
+### Enable optimized parameter binding
+Microsoft.Data.SqlClient introduces new `SqlCommand` API, `EnableOptimizedParameterBinding` to improve performance of queries with large number of parameters. This property is disabled by default. When set to `true`, parameter names will not be sent to the SQL server when the command is executed.
```
Comment (David-Engel) [inline release-notes/4.0/4.0.0.md:77]:
```suggestion
Microsoft.Data.SqlClient introduces a new `SqlCommand` API, `EnableOptimizedParameterBinding` to improve performance of queries with a large number of parameters. This property is disabled by default. When set to `true`, parameter names will not be sent to the SQL server when the command is executed.
```

File: release-notes/4.0/4.0.0.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)
+- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
+
+### Encrypt default value set to true
+The default value of the `Encrypt` connection setting has been changed from `false` to `true`. With the growing use of cloud databases and the need to ensure those connections are secure, it's time for this backwards-compatibility-breaking change.
+
+### Ensure connections fail when encryption is required
+In scenarios where client encryption libraries were disabled or unavailable, it was possible for unencrypted connections to be made when Encrypt was set to true or the server required encryption.
+
+###  App Context Switch for using System default protocols
```
Comment (DavoudEshtehari) [inline release-notes/4.0/4.0.0.md:73]:
```suggestion
### App Context Switch for using System default protocols
```

**Review by DavoudEshtehari — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Review by David-Engel — APPROVED** [review]

**Pull request merged**